common/djangoapps/course_modes/views.py

```diff
@@ -226,3 +228,31 @@ def _get_requested_mode(self, request_dict):
             return 'honor'
         else:
             return None
+
+
+@ensure_csrf_cookie
+@require_level('staff')
```

Comment:
Here's the original conversation of Nov 26th.
https://github.com/edx/edx-platform/pull/6071#discussion-diff-20940272